src/librbd/internal.cc

```diff
@@ -1700,17 +1700,21 @@ int mirror_image_disable_internal(ImageCtx *ictx, bool force,
       RWLock::WLocker snap_locker(ictx->snap_lock);
       uint64_t new_features;
       if (enabled) {
+      	if (features & ictx->features != 0) {
```

Comment:
You need to put braces around ```features & ictx->features``` -- similar comment below.  Right now the rbd CLI isn't throwing the error.